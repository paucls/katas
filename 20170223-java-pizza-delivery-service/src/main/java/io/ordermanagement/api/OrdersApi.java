package io.ordermanagement.api;

import io.ordermanagement.model.Order;
import io.swagger.annotations.*;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

@Api(value = "orders", description = "the orders API")
public interface OrdersApi {

    @ApiOperation(value = "", notes = "Creates a new `Order` resource", response = Order.class, tags = {})
    @ApiResponses(value = {@ApiResponse(code = 200, message = "Order created successfully", response = Order.class)})
    @RequestMapping(value = "/orders", method = RequestMethod.POST)
    ResponseEntity<Order> createOrder(
            @ApiParam(value = "Order to create", required = true) @RequestBody Order order
    );

}
