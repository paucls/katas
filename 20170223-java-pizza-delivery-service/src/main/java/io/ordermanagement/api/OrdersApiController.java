package io.ordermanagement.api;

import io.ordermanagement.model.Order;
import io.ordermanagement.repository.OrderRepository;
import io.swagger.annotations.ApiParam;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;

@Controller
public class OrdersApiController implements OrdersApi {

    @Autowired
    private OrderRepository orderRepository;

    @Override
    public ResponseEntity<List<Order>> getAllOrders() {
        return new ResponseEntity<>(orderRepository.findAll(), HttpStatus.OK);
    }

    public ResponseEntity<Order> createOrder(
            @ApiParam(value = "Order to create", required = true) @RequestBody Order order) {
        Order createdOrder = orderRepository.saveAndFlush(order);
        return new ResponseEntity<>(createdOrder, HttpStatus.OK);
    }

}
