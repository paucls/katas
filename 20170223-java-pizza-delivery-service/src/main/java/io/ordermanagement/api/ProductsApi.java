package io.ordermanagement.api;

import io.ordermanagement.model.Product;
import io.swagger.annotations.*;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

;

@Api(value = "products", description = "the products API")
public interface ProductsApi {

    @ApiOperation(value = "", notes = "Gets the list of all products. ", response = Product.class, responseContainer = "List", tags = {})
    @ApiResponses(value = {@ApiResponse(code = 200, message = "An array of `Product` objects", response = Product.class)})
    @RequestMapping(value = "/products", method = RequestMethod.GET)
    ResponseEntity getAllProducts();

    @ApiOperation(value = "", notes = "Creates a new product", response = Product.class, tags = {})
    @ApiResponses(value = {@ApiResponse(code = 200, message = "Product created successfully", response = Product.class)})
    @RequestMapping(value = "/products", method = RequestMethod.POST)
    ResponseEntity<Product> createProduct(
            @ApiParam(value = "Product to create", required = true) @RequestBody Product product
    );

}
