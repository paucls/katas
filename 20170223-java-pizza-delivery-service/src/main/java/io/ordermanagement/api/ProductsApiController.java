package io.ordermanagement.api;

import io.ordermanagement.model.Product;
import io.ordermanagement.repository.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;

@Controller
public class ProductsApiController implements ProductsApi {

    ProductRepository productRepository;

    @Autowired
    public ProductsApiController(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    @Override
    public ResponseEntity<List<Product>> getAllProducts() {
        return new ResponseEntity<>(productRepository.findAll(), HttpStatus.OK);
    }

    @Override
    public ResponseEntity<Product> createProduct(@RequestBody Product product) {
        Product createdProduct = productRepository.saveAndFlush(product);
        return new ResponseEntity<Product>(createdProduct, HttpStatus.CREATED);
    }

}
