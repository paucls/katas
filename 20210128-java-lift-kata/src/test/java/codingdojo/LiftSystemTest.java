package codingdojo;

import org.junit.jupiter.api.Test;

import static java.util.List.of;
import static org.approvaltests.Approvals.verify;

public class LiftSystemTest {

    @Test
    public void a_lift_should_not_move_when_there_is_no_request_for_it() {
        Lift liftA = new Lift("A", 0);
        LiftSystem lifts = new LiftSystem(of(0, 1), of(liftA), of());
        lifts.tick();
        verifyLifts(lifts);
    }

    @Test
    public void a_lift_should_move_up_to_requested_floor_and_open_doors_in_order_to_fulfill_request() {
        Lift liftA = new Lift("A", 0);
        LiftSystem lifts = new LiftSystem(of(0, 1), of(liftA), of(new Call(1, Direction.DOWN)));
        lifts.tick();
        verifyLifts(lifts);
    }

    @Test
    public void a_lift_should_move_down_to_requested_floor_and_open_doors_in_order_to_fulfill_request() {
        Lift liftA = new Lift("A", 1);
        LiftSystem lifts = new LiftSystem(of(0, 1), of(liftA), of(new Call(0, Direction.UP)));
        lifts.tick();
        verifyLifts(lifts);
    }

    @Test
    void should_not_send_request_to_lift_when_its_doors_are_open() {
        Lift liftA = new Lift("A", 0, true);
        LiftSystem lifts = new LiftSystem(of(0, 1), of(liftA), of(new Call(1, Direction.DOWN)));
        lifts.tick();
        verifyLifts(lifts);
    }

    private void verifyLifts(LiftSystem lifts) {
        verify(new LiftSystemPrinter().print(lifts));
    }
}
