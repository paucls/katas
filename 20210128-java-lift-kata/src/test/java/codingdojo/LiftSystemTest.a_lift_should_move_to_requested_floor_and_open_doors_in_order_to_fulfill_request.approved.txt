1 v   ]A[1
0        0
